src/mem/freelist.h

```diff
@@ -0,0 +1,297 @@
+#pragma once
+/**
+ * This file encapsulates the in disused object free lists
+ * that are used per slab.
```

Comment:
Perhaps "small-object Slab" not just "slab" to emphasize that this is not used for `Mediumslab`s or `Largeslab`s?